partials/expenses_input.html

```diff
@@ -14,7 +16,7 @@
                     class="steps"
                     min="0"
                     max="99999"
-                    ng-model="expenses_ctrl.expenses" />
+                    ng-model="expenses" />

```

Comment:
I think these might need to be left as values on a particular object. Angular treats double data-bound literals strangely when they get updated. Often the data binding will be lost and values will stop updating as expected. When made part of an object (like it was before) it's no longer a literal and the binding is preserved.